src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -47,6 +47,87 @@ internal class RPackageManager : IRPackageManager {
             await RPackageWebParser.RetrievePackageInfo(uri, pkg);
         }
 
+        public void InstallPackage(string name, string libraryPath) {
+            string script;
+            if (string.IsNullOrEmpty(libraryPath)) {
+                script = string.Format("install.packages(\"{0}\")", name);
+            } else {
+                script = string.Format("install.packages(\"{0}\", lib=\"{1}\")", name, libraryPath.ToRPath());
+            }
+
+            _interactiveWorkflow.Operations.EnqueueExpression(script, true);
+        }
+
+        public void UninstallPackage(string name, string libraryPath) {
+            string script;
+            if (string.IsNullOrEmpty(libraryPath)) {
+                script = string.Format("remove.packages(\"{0}\")", name);
+            } else {
+                script = string.Format("remove.packages(\"{0}\", lib=\"{1}\")", name, libraryPath.ToRPath());
+            }
+
+            _interactiveWorkflow.Operations.EnqueueExpression(script, true);
+        }
+
+        public void LoadPackage(string name, string libraryPath) {
+            string script;
+            if (string.IsNullOrEmpty(libraryPath)) {
+                script = string.Format("library(\"{0}\")", name);
+            } else {
+                script = string.Format("library(\"{0}\", lib.loc=\"{1}\")", name, libraryPath.ToRPath());
+            }
+
+            _interactiveWorkflow.Operations.EnqueueExpression(script, true);
+        }
+
+        public void UnloadPackage(string name) {
+            string script = string.Format("detach(\"package:{0}\", unload=TRUE)", name);
+
+            _interactiveWorkflow.Operations.EnqueueExpression(script, true);
+        }
+
+        public async Task<string[]> GetLoadedPackagesAsync() {
+            if (!_interactiveWorkflow.RSession.IsHostRunning) {
+                return new string[0];
+            }
+
+            try {
```

Comment:
I have a message bar in the package manager that can show errors, but I haven't wired it to any error handling yet. Shouldn't be hard though. 
@huguesv, do you have a list of expected exceptions?